ip6.h

```diff
@@ -197,6 +197,15 @@ struct ip6_srh {
 	nd_ipv6		srh_segments[1];	/* SRH segments list*/
 };
 
+/**
+ * Type 0 => Pad1
+ * Type 4 => PadN
+ * Type 5 => HMAC
+ */
+#define IPV6_RTHDR_TLV_TYPE_0 0
+#define IPV6_RTHDR_TLV_TYPE_4 4
+#define IPV6_RTHDR_TLV_TYPE_5 5
+
```

Comment:
These TLVs are SRH specific so I would suggest to rename them. 
Also you could use the name of the TLV in the definition instead of just using TYPE_0, TYPE_4,... 

I would do: 


#define IPV6_SRH_TLV_PAD1    0 
#define IPV6_SRH_TLV_PADN   4
#define IPV6_SRH_TLV_HMAC   5
#define IPV6_RTHDR_TLV_TYPE_4 4
 #define IPV6_RTHDR_TLV_TYPE_5 5